lib/active_model/serializer/adapter.rb

```diff
@@ -11,6 +12,47 @@ class Adapter
 
       attr_reader :serializer
 
+      # Only the Adapter class has these methods.
+      # None of the sublasses have them.
+      class << ActiveModel::Serializer::Adapter
+        def adapter_map
+          ADAPTER_MAP
+        end
+
+        def register(name, klass)
+          adapter_map.update(name.to_s.underscore => klass)
+        end
+
+        def get(adapter)
```

Comment:
Yeap, it's exactly the same logic. IMHO it's related to this PR, check this out:

We can replace the actual code:
```ruby
def self.adapter
  adapter_class = case config.adapter
  when Symbol
    ActiveModel::Serializer::Adapter.adapter_class(config.adapter)
  when Class
    config.adapter
  end
  unless adapter_class
    valid_adapters = Adapter.constants.map { |klass| ":#{klass.to_s.downcase}" }
    raise ArgumentError, "Unknown adapter: #{config.adapter}. Valid adapters are: #{valid_adapters}"
  end

  adapter_class
end
```

By something like:
```ruby
def self.adapter
  ActiveModel::Serializer::Adapter.adapter_class(config.adapter)
end
```